### ISSUE TITLE
thread 'main' panicked at src/adapter/src/catalog/open.rs:534:21: internal error: failed to load bootstrap view: pg_catalog.pg_aggregate error: PlanError(Unstructured("CAST does not support casting from text to regproc"))

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#26685]
### What version of Materialize are you using?

c189146fd0 (Pull Request #26665)

### What is the issue?

Seen in https://buildkite.com/materialize/nightly/builds/7464#018eebef-06d8-4040-8c27-9f9aa7f7c3cf
```
thread 'main' panicked at src/adapter/src/catalog/open.rs:534:21:
internal error: failed to load bootstrap view:
pg_catalog.pg_aggregate
error:
PlanError(Unstructured("CAST does not support casting from text to regproc"))

Make sure that:

                        - the schema name is specified in the builtin view's create sql statement
                        - any used function or cast implementations that are SQL are special cased in `visit_system_view_dependency_names` (see the notes there about reg* casts)
                        
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: <mz_adapter::catalog::Catalog>::parse_views::{closure#0}::{closure#0}
   3: <mz_adapter::catalog::Catalog>::initialize_state::{closure#0}
   4: <tracing::instrument::Instrumented<<mz_adapter::catalog::Catalog>::initialize_state::{closure#0}> as core::future::future::Future>::poll
   5: <tracing::instrument::Instrumented<<mz_adapter::catalog::Catalog>::open::{closure#0}> as core::future::future::Future>::poll
   6: <mz_testdrive::action::State>::with_catalog_copy::<mz_testdrive::action::consistency::check_catalog_state::{closure#0}::{closure#0}, mz_adapter::catalog::state::CatalogState>::{closure#0}
   7: mz_testdrive::run_line_reader::{closure#0}
   8: mz_testdrive::run_string::{closure#0}
   9: testdrive::main::{closure#0}
  10: <tracing::instrument::Instrumented<testdrive::main::{closure#0}> as core::future::future::Future>::poll
  11: <tokio::runtime::park::CachedParkThread>::block_on::<tracing::instrument::Instrumented<testdrive::main::{closure#0}>>
  12: <tokio::runtime::runtime::Runtime>::block_on::<testdrive::main::{closure#0}>
  13: testdrive::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
Note that the panic is in testdrive, so I'm not sure if this is actually a release blocker, but it could be, so marking as one just in case.
Probably unrelated to the PR where it occurred: https://github.com/MaterializeInc/materialize/pull/26665
I think this was introduced in https://github.com/MaterializeInc/materialize/pull/26634

**Comment by danhhz** [comment]
Looks like https://github.com/MaterializeInc/materialize/pull/26634 isn't in v0.96.x, so if that's indeed the cause, not a blocker for today's rollout, but a blocker for the candidate cut tomorrow

**Comment by maddyblue** [comment]
I was able to reproduce this locally by introducing a random delay into parse_item. This could happen in any startup of envd. It however might not need to be a release blocker because if it does happen, the best workaround is to just start envd again. Investigating a full fix, but might not need to prevent anything from going out?

**Comment by maddyblue** [comment]
Update: I tried to improve the dependency calculator. I found tons of places where we had omitted schemas, but we also omit them in various function defs all over too. I now believe the current method is fundamentally flawed and needs replacing (should use error sniffing with retries which we've done before). I will not have a PR in by the commit cut for this.

I *still* think it's ok to remove the release-blocker tag from this issue because:
- this bug can only happen on startup
- it happens rarely (we've only seen it one time so far in nightlies)
- the workaround is to try starting envd again

We can remove the tag if I can get one other person to agree with that rationale.
